objects/69ead2/contained/DUESETUP/script.lua

```diff
@@ -0,0 +1,643 @@
+spiritName = "Dances Up Earthquakes"
+
+energyPoint = Vector(-0.99, 0.20, -0.41) -- location of the snap point where the presence for 2 impending energy/turn sits
+impendTable = {}
+g3Selected = false
+
+aboveCard = 0.5
+buttonVisibility = 0.4 -- increase for testing, 0 ultimately. May put this value dependent on a slider so people can vary?
+
+------------------
+-- setup functions
+------------------
+
+function onSave()
+    local json = JSON.decode(self.script_state)
+    if not json then
+        json = {}
+    end
+    json.impendTable = impendTable
+    json.g3Selected = g3Selected
+    return JSON.encode(json)
+end
+
+function updateSave()
+    local json = JSON.decode(self.script_state)
+    if not json then
+        json = {}
+    end
+    json.impendTable = impendTable
+    json.g3Selected = g3Selected
+    self.script_state = JSON.encode(json)
+end
+
+function onLoad(saved_data)
+    -- this just makes the letter "i" above this object, if visuals change in any way, this can be removed
+    self.createButton({
+        click_function = "null",
+        function_owner = self,
+        label = "i",
+        position = {0,0.3,0.1},
+        width = 0,
+        height = 0,
+        font_size = 450,
+        font_color = {1,1,1},
+    })
+
+    if saved_data ~= "" then
+        local loaded_data = JSON.decode(saved_data)
+        setupComplete = loaded_data.setupComplete
+        impendTable = loaded_data.impendTable
+        g3Selected = loaded_data.g3Selected
+    end
+
+    if setupComplete then
+        createPanelButtons()
+        createAllCardButtons()
+    end
+end
+
+function null()
+end
+
+function doSetup(params)
+    local color = params.color
+    local panel = Global.call("getSpirit", {name = spiritName})
+    self.locked = true
+    self.addTag("Modify Cost")
```

Comment:
This was for fixing something funky that was occurring back when this was initially introduced. Unsure on the specifics any more and I've just tested and it seems ok to have it tagged initially now. I'll make the change but it may be worth keeping this in mind in case it is something that breaks under niche criteria.